src/analysis/processing/qgsalgorithmrandompointsonlines.cpp

```diff
@@ -0,0 +1,264 @@
+  if ( !sink )
+    throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+
+  //initialize random engine
+  srand(mRandSeed);
+
+  //index for finding nearest neighbours (mMinDistance > 0)
+  QgsSpatialIndex index = QgsSpatialIndex();
+
+  int totNPoints = 0;
+  int missedPoints = 0;
+  int missedLines = 0;
+  int emptyOrNullGeom = 0;
+
+  int featureCount = 0;
+  QgsFeature lFeat;
+  QgsFeatureIterator fitL = lineSource->getFeatures();
+  while ( fitL.nextFeature( lFeat ) )
+  {
+    featureCount++;
+    if ( feedback->isCanceled() )
+    {
+      feedback->pushInfo("cancelled");
+      break;
+    }
+    if ( not lFeat.hasGeometry() )
+    {
+      // Increment invalid features count
+      feedback->pushInfo("No geometry");
+      emptyOrNullGeom++;
+      continue;
+    }
+    QgsGeometry lGeom( lFeat.geometry() );
+    if (lGeom.isNull())
+    {
+      // Increment invalid features count
+      feedback->pushInfo("Null geometry");
+      emptyOrNullGeom++;
+      continue;
+    }
+    if (lGeom.isEmpty())
+    {
+      // Increment invalid features count
+      feedback->pushInfo("Empty geometry");
+      emptyOrNullGeom++;
+      continue;
+    }
+    float lineLength = lGeom.length();
+    int distCheckIterations = 0;
+    int i = 0;
+    while ( i < mNumPoints and distCheckIterations < mMaxAttempts )
+    {
+      if ( feedback->isCanceled() )
+      {
+        break;
+      }
+
+      float randPos = lineLength * (float) rand() / RAND_MAX;
+      QgsGeometry rpGeom = QgsGeometry( lGeom.interpolate( randPos ) );
+      //if ( not rpGeom.isNull() and not rpGeom.isEmpty() )
+      if ( rpGeom.isNull() or rpGeom.isEmpty() )
+      {
+        distCheckIterations++;
+        continue;
+      }
+
+      QgsPointXY rPoint;
+      rPoint = QgsPoint( *qgsgeometry_cast< const QgsPoint * >( rpGeom.constGet() ) );
+
+      if ( not mMinDistance == 0 )
+      {
+        if ( i > 0 )
+        {
+          QList<QgsFeatureId> neighbors = index.nearestNeighbor( rPoint, 1, mMinDistance );
+          if (not neighbors.empty() )
+          {
+            distCheckIterations++;
+            continue;
+          }
+        }
+      }
+      QgsFeature f = QgsFeature();
+      QgsAttributes pAttrs = QgsAttributes();
+      pAttrs.append( totNPoints );
+      if ( mIncludeLineAttr ) {
+          pAttrs.append( lFeat.attributes() );
+      }
+      f.setAttributes( pAttrs );
+      f.setGeometry( rpGeom );
+      if ( not mMinDistance == 0 )
+      {
+        index.addFeature( f );
+      }
+      sink->addFeature( f, QgsFeatureSink::FastInsert );
+      totNPoints++;
+      i++;
+      distCheckIterations = 0; //reset distCheckIterations if a point is added
+      feedback->setProgress( static_cast<int>( static_cast<double>( totNPoints ) / static_cast<double>( mNumPoints ) * 100 ) );
```

Comment:
Progress should be reported based on the current line feature vs overall feature count, otherwise the progress will keep resetting back to zero as each new line feature is processed

Reply:
The progress reporting was very messy. It should at least be pretty consistent now.